engine/apps/slack/scenarios/shift_swap_requests.py

```diff
@@ -0,0 +1,201 @@
+import json
+import logging
+import typing
+
+from apps.slack.constants import DIVIDER
+from apps.slack.models import SlackMessage
+from apps.slack.scenarios import scenario_step
+from apps.slack.types import Block, BlockActionType, EventPayload, PayloadType, ScenarioRoute
+
+if typing.TYPE_CHECKING:
+    from apps.schedules.models import ShiftSwapRequest
+    from apps.slack.models import SlackTeamIdentity, SlackUserIdentity
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+SHIFT_SWAP_PK_ACTION_KEY = "shift_swap_request_pk"
+
+
+class BaseShiftSwapRequestStep(scenario_step.ScenarioStep):
+    def _generate_blocks(self, shift_swap_request: "ShiftSwapRequest") -> Block.AnyBlocks:
+        pk = shift_swap_request.pk
+
+        # TODO: come up with a better layout for this..
+        main_message_text = f"Your teammate {shift_swap_request.beneficiary.get_username_with_slack_verbal()} has submitted a shift swap request."
+
+        blocks: Block.AnyBlocks = [
+            typing.cast(
+                Block.Section,
+                {
+                    "type": "section",
+                    "text": {
+                        "type": "mrkdwn",
+                        "text": main_message_text,
+                    },
+                },
+            ),
+            typing.cast(
+                Block.Section,
+                {
+                    "type": "section",
+                    "text": {
+                        "type": "mrkdwn",
+                        # TODO: I believe it'll be easier to wait to generate this until we have the schedule override changes in place
+                        # NOTE: use apps.slack.utils.format_datetime_to_slack method to format the datetimes
+                        "text": "*📅 Shift Details*: 9h00 - 17h00 (UTC) daily from Monday July 24, 2023 - July 28, 2023",
```

Comment:
note. this is purposely being hardcoded for now. We need to first finish the logic which generates the overrides before we can determine this. We'll follow up in a future (and smaller 🙈) PR to update this